production/helm/loki/templates/memcached/_memcached-statefulset.tpl

```diff
@@ -122,8 +122,12 @@ spec:
             {{- end }}
           securityContext:
             {{- toYaml $.ctx.Values.memcached.containerSecurityContext | nindent 12 }}
-          {{- if .extraVolumeMounts }}
           volumeMounts:
+          {{- if .persistence.enabled }}
+            - name: data
+              mountPath: {{ .persistence.mountPath }}
+          {{- end }}
+          {{- if .extraVolumeMounts }}
             {{- toYaml .extraVolumeMounts | nindent 12 }}
           {{- end }}
```

Comment:
How does this behave when there are neither `.extraVolumeMounts` and `.persistence.enabled` is `false`? I would assume it renders the key `volumeMounts:` but without list items. I assume this would cause invalid YAML.
In that case, it should either not render `volumeMounts` at all, or `volumeMounts: []`.

Reply:
> I would assume it renders the key volumeMounts: but without list items

that is correct

> I assume this would cause invalid YAML.

that is not an issue, for exemple [env](https://github.com/grafana/loki/blob/fd12ba3b0ee9cea844edb86afdcc93453b72475d/production/helm/loki/templates/memcached/_memcached-statefulset.tpl#L115-L118) is [empty by default](https://github.com/grafana/loki/blob/fd12ba3b0ee9cea844edb86afdcc93453b72475d/production/helm/loki/values.yaml#L1) but I've added another check just in case

Reply:
Hm, interesting, I would expect that to be invalid. Nevertheless, thanks for the additional check.